python/sauron-core/core/analysis/stats.py

```
@@ -0,0 +1,197 @@
+                if np.isnan(metrics[key]) or np.isinf(metrics[key]):
+                    metrics[key] = 0.0
+                else:
+                    # Clip extreme values to reasonable range
+                    metrics[key] = float(np.clip(metrics[key], -1e10, 1e10))
+
+            # Compute final score
+            metrics['ai_score'] = self._compute_ai_probability(metrics)
+
+            return metrics
+
+        except Exception as e:
```

Ruff (0.8.2)

Do not catch blind exception: `Exception`

(BLE001)